app/fields/em_matrix/EmMatrixField.php

```diff
@@ -7,20 +7,44 @@ class EmMatrixField extends FieldBase
 	 */
 	public function getValue()
 	{
-		if(empty($this->fieldValue))
+		$select  = [
+			'from'  => $this->settings['nodeTableCode'],
+			'where' => [
+				'operation' => 'and',
+				'fields'    => [[
+					'code'      => $this->settings['nodeField'],
+					'operation' => 'IS',
+					'value'     => $this->fieldValue
+				]]
+			]
+		];
+		$node = $this->element->select($select);
+
+		if(!count($node) )
 			return [];
 
-		return explode(',', $this->fieldValue);
+		return [
+			'matrixValue' => $node
+		];
 	}
 
 	/**
 	 * Сохранить значение
 	 */
 	public function saveValue()
 	{
-		if(is_array($this->fieldValue))
-			return implode(',',$this->fieldValue);
-		else
-			return '';
+		if(empty($this->fieldValue))
+			return NULL;
+
+		if(is_numeric($this->fieldValue))
+			return intval($this->fieldValue);
+
+		if(is_string($this->fieldValue))
+			return NULL;
+
+		if(!is_array($this->fieldValue))
+			return intval($this->fieldValue);
+
```

Comment:
```suggestion
		if(empty($this->fieldValue))
			return NULL;

		if(is_numeric($this->fieldValue))
			return intval($this->fieldValue);

		if(is_string($this->fieldValue))
			return NULL;

		if(!is_array($this->fieldValue))
			return intval($this->fieldValue);

```
можно превратить в два if